driver-core/src/test/unit/com/mongodb/connection/ClusterSettingsSpecification.groovy

```diff
@@ -288,15 +288,16 @@ class ClusterSettingsSpecification extends Specification {
         settings.hosts == [new ServerAddress('example.com:27018')]
     }
 
-    def 'when cluster type is unknown and replica set name is specified, should set cluster type to ReplicaSet'() {
+    def 'when cluster type is UNKNOWN and replica set name is set, should set cluster type to REPLICA_SET and mode to MULTIPLE'() {
```

Comment:
Consider adding more tests which combine connection string application with explicit use of the builder properties (both before and after connection string application), e.g.

```
ClusterSettings.builder()
.applyConnectionString(new ConnectionString("mongodb://127.0.0.1:27017/"))
.requiredReplicaSetName("replset")
.build()
.getMode()
```

and

```
ClusterSettings.builder()
.requiredReplicaSetName("replset")
.applyConnectionString(new ConnectionString("mongodb://127.0.0.1:27017/"))
.build()
.getMode()
```